### PR TITLE
maintainer_gh.pl - fix for making it work on dev instances

### PR DESCRIPTION
**Pull request by MariagraziaAlastra**
@jdorweiler was having issues updating his data, since he has some usernames from prod, but obviously doesn't have the actual users on his instance's DB, so I made some tweaks to the maintainer script!

//cc @zachthompson

### CODE REVIEW

File: sql/1.145/maintainer_gh.pl [hunk 1]
```diff
@@ -14,10 +14,10 @@
         my %updated_maintainer;
         my $duckco_user = $d->rs('User')->find({username => $maintainer});
 
-        if ($duckco_user) {
+        if ($duckco_user || (!$d->is_live)) {
```
Comment (zachthompson) [inline sql/1.145/maintainer_gh.pl]: Outside of the `while` loop, i.e. line 11:
```Perl
my $is_live = $d->is_live;
```
Then use the variable.

Won't this set the `duckco` maintainer to a `github` maintainer if the user doesn't exist?

Comment (MariagraziaAlastra) [inline sql/1.145/maintainer_gh.pl]: @zachthompson updated, thanks!

File: sql/1.145/maintainer_gh.pl [hunk 2]
```diff
@@ -3,30 +3,37 @@
...
+            my $duckco_user = $d->rs('User')->find({username => $maintainer});
 
-        if ($duckco_user) {
-            %updated_maintainer = ( duckco => $maintainer );
+            if ($duckco_user || (!$is_live)) {
+                %updated_maintainer = ( duckco => $maintainer );
 
-            if ($duckco_user->github_id) {
-                $updated_maintainer{github} = $d->rs('GitHub::User')->find({github_id => $duckco_user->github_id})->login;
+                if ($is_live && $duckco_user->github_id) {
```
Comment (zachthompson) [inline sql/1.145/maintainer_gh.pl]: I don't think we need `$is_live` here.  I have tEStfOUr set up to use my git username and it didn't update correctly.  We should be able to do:
```Perl
if(my $git_id = $duckco_user->github_id){
....
```

Comment (MariagraziaAlastra) [inline sql/1.145/maintainer_gh.pl]: Right, I didn't consider that. 
Done, thanks!

**Comment by zachthompson** [comment]
Thank you.

**Comment by MariagraziaAlastra** [comment]
Thank you! :cactus:

**Pull request merged**